test/HypLSP8.t.sol

```diff
@@ -0,0 +1,218 @@
+import { HypLSP8Collateral } from "../src/HypLSP8Collateral.sol";
+import { LSP8Mock } from "./LSP8Mock.sol";
+
+abstract contract HypTokenTest is Test {
+    using TypeCasts for address;
+
+    uint256 internal constant INITIAL_SUPPLY = 10;
+    string internal constant NAME = "Hyperlane NFTs";
+    string internal constant SYMBOL = "HNFT";
+
+    address internal ALICE = makeAddr("alice");
+    address internal BOB = makeAddr("bob");
+    address internal OWNER = makeAddr("owner");
+    uint32 internal constant ORIGIN = 11;
+    uint32 internal constant DESTINATION = 22;
+    bytes32 internal constant TOKEN_ID = bytes32(uint256(1));
+    string internal constant URI = "http://example.com/token/";
+
+    LSP8Mock internal localPrimaryToken;
+    LSP8Mock internal remotePrimaryToken;
+    TestMailbox internal localMailbox;
+    TestMailbox internal remoteMailbox;
+    TokenRouter internal localToken;
+    HypLSP8 internal remoteToken;
+    TestPostDispatchHook internal noopHook;
+
+    function setUp() public virtual {
+        localMailbox = new TestMailbox(ORIGIN);
+        remoteMailbox = new TestMailbox(DESTINATION);
+
+        localPrimaryToken = new LSP8Mock(NAME, SYMBOL, OWNER);
+
+        noopHook = new TestPostDispatchHook();
+        localMailbox.setDefaultHook(address(noopHook));
+        localMailbox.setRequiredHook(address(noopHook));
+
+        remoteToken = new HypLSP8(address(remoteMailbox));
+
+        vm.prank(OWNER);
+        remoteToken.initialize(0, NAME, SYMBOL);
+
+        vm.deal(ALICE, 1 ether);
+    }
+
+    function _deployRemoteToken(bool isCollateral) internal {
+        if (isCollateral) {
+            remoteToken = new HypLSP8(address(remoteMailbox));
+            vm.prank(OWNER);
+            remoteToken.initialize(0, NAME, SYMBOL);
+        } else {
+            remoteToken = new HypLSP8(address(remoteMailbox));
+            vm.prank(OWNER);
+            remoteToken.initialize(0, NAME, SYMBOL);
+        }
+        vm.prank(OWNER);
+        remoteToken.enrollRemoteRouter(ORIGIN, address(localToken).addressToBytes32());
+    }
+
+    function _processTransfers(address _recipient, bytes32 _tokenId) internal {
+        vm.prank(address(remoteMailbox));
+        remoteToken.handle(
+            ORIGIN, address(localToken).addressToBytes32(), abi.encodePacked(_recipient.addressToBytes32(), _tokenId)
+        );
+    }
+
+    function _performRemoteTransfer(uint256 _msgValue, bytes32 _tokenId) internal {
+        vm.prank(ALICE);
+        localToken.transferRemote{ value: _msgValue }(DESTINATION, BOB.addressToBytes32(), uint256(_tokenId));
+
+        _processTransfers(BOB, _tokenId);
+        assertEq(remoteToken.balanceOf(BOB), 1);
+    }
+}
+
+contract HypLSP8Test is HypTokenTest {
+    using TypeCasts for address;
+
+    HypLSP8 internal lsp8Token;
+
+    function setUp() public override {
+        super.setUp();
+
+        localToken = new HypLSP8(address(localMailbox));
```

Comment:
What do these two lines do? I am not sure to understand. What is the difference between `localToken` and `lsp8Token`?

Reply:
localToken and lsp8Token point to the same contract instance but localToken is typed as the parent TokenRouter for base functionality while lsp8Token is typed as HypLSP8 to access LSP8-specific functions.